api/server/httputils/errors.go

```diff
@@ -85,9 +85,11 @@ func apiVersionSupportsJSONErrors(version string) bool {
 
 // MakeErrorHandler makes an HTTP handler that decodes a Docker error and
 // returns it in the response.
-func MakeErrorHandler(err error) http.HandlerFunc {
+func MakeErrorHandler(err error, statusCode int) http.HandlerFunc {
```

Comment:
discussing with @tiborvass  - don't think we really need the `statusCode` argument here, as it's only to optimize (not having to call `GetHTTPErrorStatusCode`twice)